api/v1/cluster_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+MIT License
+
+Copyright (c) 2021 StorageOS
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterSpec defines the desired state of Cluster.
+type ClusterSpec struct {
+	// LicenseKey is the cluster license key.
+	LicenseKey []byte `json:"licenseKey,omitempty"`
+	// DisableTelemetry is used to configure telemetry.
+	DisableTelemetry bool `json:"disableTelemetry,omitempty"`
+	// DisableCrashReporting is used to configure crash reporting.
+	DisableCrashReporting bool `json:"disableCrashReporting,omitempty"`
+	// DisableVersionCheck is used to configure version check.
+	DisableVersionCheck bool `json:"disableVersionCheck,omitempty"`
+	// LogLevel is used to set the log level.
+	LogLevel string `json:"logLevel,omitempty"`
+	// LogFormat is the format of the cluster logs.
+	LogFormat string `json:"logFormat,omitempty"`
+	// Cert is a x509-encoded certificate.
+	Cert []byte `json:"cert,omitempty"`
+	// Key is a x509-encoded key.
+	Key []byte `json:"key,omitempty"`
+}
+
+// ClusterStatus defines the observed state of Cluster.
+type ClusterStatus struct{}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:scope=Cluster
+
+// Cluster is the Schema for the clusters API.
+type Cluster struct {
+	metav1.TypeMeta   `json:",inline"`
```

Comment:
Do we need the comma?

Reply:
Yes, that's not the json field name and comma is needed to separate the name and option.
This is a generated code, btw.